### ISSUE TITLE
[2022.5] CalendarEventDevice is deprecated, modify ICalCalendarEventDevice to extend CalendarEntity

### ISSUE THREAD

**Issue by golles** [tybritten/ical-sensor-homeassistant#91]
Just updated HA to 2022.5 and get the following warning:
```
2022-05-04 20:46:20 WARNING (MainThread) [homeassistant.components.calendar] CalendarEventDevice is deprecated, modify ICalCalendarEventDevice to extend CalendarEntity
```

More info on the dev blog: https://developers.home-assistant.io/blog/2022/04/18/calendar-data-model/

Thank you!

**Comment by tiaanv** [comment]
Is this going to be addressed?  Makes the component a little useless in terms of standard calendar automations...

**Comment by tiaanv** [comment]
It ended up being a few small changes to get this working, and in line with the new CalendarEntity Requirements.

**Changes required:**
- Change the Entity Type to the new CalendayEntity
- change the function that gets the list of entries to return a strongly typed class, rather than a dictionary
- Change the Event (current) returned, also to a strongly typed class.

@tybritten There are probably better ways of cleaning things up, but the way I did it, is the least amount of changes to your code.

_I tested the key reason I needed this.  Using the calendar as a trigger for automation. It worked perfectly._

**Instructions:**
Simply replace the two files in your custom components/ical folder with the files in the zip file attached.
[calendar.zip](https://github.com/tybritten/ical-sensor-homeassistant/files/9128090/calendar.zip)

**Comment by tybritten** [comment]
Fixes Mereged. Thanks @tiaanv

**Comment by Olen** [comment]
A beta release that addresses this is now available: [v1.10.0b1](https://github.com/tybritten/ical-sensor-homeassistant/releases/tag/v1.10.0b1)

All deprecated APIs have been removed, including the migration from `Entity` to `SensorEntity` and removal of `CONNECTION_CLASS`, `CONFIG_SCHEMA`, and `setup()`.

To test: in HACS, enable "Show beta versions" in the download dialog and select `v1.10.0b1`. Feedback welcome.
